src/Symfony/Component/Serializer/Normalizer/DateTimeNormalizer.php

```diff
@@ -50,6 +52,11 @@ public function __construct($defaultContext = [], \DateTimeZone $timezone = null
             $defaultContext[self::TIMEZONE_KEY] = $timezone;
         }
 
+        // Handle legacy mixed with $defaultContext[]
```

Comment:
I don't get this. Is this a bug fix? If so it should go in another PR targeting the branch were it was introduced.

Reply:
No unfortunately.  If the user sets the construct input to
> __construct(['preserve_context_timezone' => true], 'Australia/Sydney') 

then the timezone is ignored without this code.  I think it should throw a deprecation.  In 3.4 the __construct was (legacy)
>__construct('Y.m.d H:i:s', 'Europe/Paris');
Hence the legacy.

Reply:
I think the issue here is that the bug fix has morphed into a strategy to move the default behaviour of the denormaliser from ignoring the context_timezone, to always applying the context_timezone.  As such, the system MUST handle BC so existing use cases continue to work.  DefaultContext was introduced in 4.2